src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1630,37 +1636,71 @@ void Syscollector::syncLoop(std::unique_lock<std::mutex>& lock)
         sync();
     }
 
-    while (!m_cv.wait_for(lock, std::chrono::seconds{m_intervalValue}, [&]()
+    while (!m_cv.wait_for(lock, std::chrono::seconds{m_adjustedIntervalValue ? m_adjustedIntervalValue : m_intervalValue}, [&]()
 {
-    return m_stopping;
-}))
+    return (m_stopping);
+    }))
     {
-        scan();
-        sync();
+        if (!m_lastScanOnDemand)
+        {
+            scan();
+            sync();
+            m_adjustedIntervalValue = 0;
+        }
+        else
+        {
+            m_logFunction(SYS_LOG_DEBUG_VERBOSE, "Syscollector scan interval adjusted due to a recent on demand scan.");
+            std::time_t current_time = std::time(nullptr);
+            m_adjustedIntervalValue = m_intervalValue - (current_time - m_lastScanOnDemandTime);
```

Comment:
This is OK as we decided to adjust the next scan-by-interval time based on the last on-demand scan. 👌